pkg/cluster/cluster_info_scraper.go

```diff
@@ -177,7 +178,7 @@ func (s *ClusterScraper) GetKubernetesServiceID() (svcID string, err error) {
 	return
 }
 
-func (s *ClusterScraper) GetRunningPodsOnNodes(nodeList []*api.Node) []*api.Pod {
+func (s *ClusterScraper) GetRunningAndReadyPodsOnNodes(nodeList []*api.Node) []*api.Pod {
```

Comment:
Another potential problem: we get the list of running pods first, then get the container metrics.
If during the interval, a pod goes wrong, then there may be no metrics for the containers.

So we may need to remove the container/applications during building the entityDTOs:
[container bought commodities](https://github.com/turbonomic/kubeturbo/blob/e4b9156c5838631308e96fc75073bc9d8ca8375a/pkg/discovery/dtofactory/container_dto_builder.go#L161)

[application bought commodities](https://github.com/turbonomic/kubeturbo/blob/e4b9156c5838631308e96fc75073bc9d8ca8375a/pkg/discovery/dtofactory/application_entity_dto_builder.go#L195)